packages/common-scripts/src/anyone_can_pay.ts

```diff
@@ -0,0 +1,549 @@
+      )
+    ) {
+      const cellCapacity: bigint = BigInt(output.cell_output.capacity);
+      const availableCapacity: bigint =
+        cellCapacity - minimalCellCapacity(output);
+      // should maintain minimal cell capcity in anyone-can-pay output
+      const deductCapacity: bigint =
+        capacity >= availableCapacity ? availableCapacity : capacity;
+      capacity -= deductCapacity;
+      output.cell_output.capacity =
+        "0x" + (cellCapacity - deductCapacity).toString(16);
+    }
+  }
+  // Remove all output cells with capacity equal to 0
+  txSkeleton = txSkeleton.update("outputs", (outputs) => {
+    return outputs.filter(
+      (output) => BigInt(output.cell_output.capacity) !== BigInt(0)
+    );
+  });
+
+  const getInputKey = (input: Cell) =>
+    `${input.out_point!.tx_hash}_${input.out_point!.index}`;
+  if (capacity > 0n) {
+    const changeCell: Cell = {
+      cell_output: {
+        capacity: "0x0",
+        lock: fromScript,
+        type: undefined,
+      },
+      data: "0x",
+      out_point: undefined,
+      block_hash: undefined,
+    };
+    let changeCapacity = 0n;
+    const minimalChangeCapacity: bigint = minimalCellCapacity(changeCell);
+    const firstInjectInputIndex = txSkeleton.get("inputs").size;
+
+    let previousInputs = Set<string>();
+    for (const input of txSkeleton.get("inputs")) {
+      previousInputs = previousInputs.add(getInputKey(input));
+    }
+
+    // Are all from same lock script and type script, so only need one change cell
+    for await (const inputCell of cellCollector.collect()) {
+      if (previousInputs.has(getInputKey(inputCell))) {
+        continue;
+      }
+
+      txSkeleton = txSkeleton.update("inputs", (inputs) => {
+        return inputs.push(inputCell);
+      });
+      txSkeleton = txSkeleton.update("witnesses", (witnesses) => {
+        return witnesses.push("0x");
+      });
+
+      const inputCapacity = BigInt(inputCell.cell_output.capacity);
+      let deductCapacity = inputCapacity;
+      if (deductCapacity > capacity) {
+        deductCapacity = capacity;
+      }
+      capacity -= deductCapacity;
+      changeCapacity += inputCapacity - deductCapacity;
+      if (capacity === 0n && changeCapacity >= minimalChangeCapacity) {
+        break;
+      }
+
+      changeCell.cell_output.capacity = "0x" + changeCapacity.toString(16);
+      txSkeleton = txSkeleton.update("outputs", (outputs) => {
+        return outputs.push(changeCell);
+      });
+    }
+
+    if (capacity > 0n || changeCapacity < minimalChangeCapacity) {
+      throw new Error(`Not enough capacity in from address!`);
+    }
+
+    changeCell.cell_output.capacity = "0x" + changeCapacity.toString(16);
+    txSkeleton = txSkeleton.update("outputs", (outputs) => {
+      return outputs.push(changeCell);
+    });
+
+    txSkeleton = await setupInputCell(txSkeleton, firstInjectInputIndex, {
+      config,
+    });
+  }
+
+  return txSkeleton;
+}
+
+export async function transfer(
```

Comment:
I feel like maybe we should not provide transfer here. I do want to distinguish transfer from withdraw in lumos:

* transfer is an action that you exact CKBs from a bunch of cells that are alike, it does not matter which cell is in fact used.
* withdraw is an action that you exact CKBs from exactly one cell, and you can only use that single cell.

I feel like in ACP, it's more likely we are dealing with withdrawing operation, not transfer operation.

Reply:
It's this means we can withdraw a cell with withdraw capacity range `[0,cell.capcaity-minimalCellCapcity] ∪ cell.capcaity`, and withdraw amount (from data) range `[0, cell.amount]`

Reply:
Yeah I think that's the plan.

Reply:
ok

Reply:
Updated.